ion-schema/src/system.rs

```diff
@@ -437,16 +438,13 @@ impl PendingTypes {
 
 /// Represents an array of BuiltIn derived ISL types
 /// for more information: https://amazon-ion.github.io/ion-schema/docs/isl-1-0/spec#type-system
-static DERIVED_ISL_TYPES: [&str; 10] = [
+static DERIVED_ISL_TYPES: [&str; 9] = [
     "type::{ name: lob, one_of: [ blob, clob ] }",
     "type::{ name: number, one_of: [ decimal, float, int ] }",
     "type::{ name: text, one_of: [ string, symbol ] }",
     "type::{ name: $lob, one_of: [ $blob, $clob ] }",
     "type::{ name: $number, one_of: [ $decimal, $float, $int ] }",
     "type::{ name: $text, one_of: [ $string, $symbol ] }",
-    // this is just a place holder for document type,
-    // IonSchemaElement::Document(_) type is used to verify the correctness on the validation side
-    "type::{ name: document }",
```

Comment:
🗺️ `document` is an "atomic" type instead of a "derived" type now that we have `IonSchemaElementType::Document`.